modules/core/src/main/java/org/apache/ignite/internal/processors/platform/utils/PlatformUtils.java

```diff
@@ -989,7 +990,9 @@ private static boolean knownMap(Object o) {
      * @return Result.
      */
     public static Object[] unwrapBinariesInArray(Object[] arr) {
-        Object[] res = new Object[arr.length];
+        Class<?> compType = arr.getClass().getComponentType();
+
+        Object[] res = (Object[])Array.newInstance(compType, arr.length);
```

Comment:
Great! It was strange, that here was untyped array. Great catch!

Reply:
Now that we have a check for component type above, this can be changed back to `Object[] res = new Object[arr.length];`